packages/http-client-java/generator/http-client-generator-core/src/main/java/com/microsoft/typespec/http/client/generator/core/mapper/PagingMetadata.java

```diff
@@ -34,18 +38,21 @@ final class PagingMetadata {
     private final ModelPropertySegment nextLinkPropertyReference;
     private final IType lroPollResultType;
     private final List<ClientMethod> nextMethods;
+    private final ClientMethodParameter maxPageSizeParameter;
     private final MethodPageDetails.ContinuationToken continuationToken;
 
     /**
      * Creates paging meta data for an {@link Operation}.
      *
      * @param operation the operation to create the paging metadata for.
-     * @param proxyMethod the proxy method representing the paged proxy api.
+     * @param parametersDetails the details about all parameters of the page api.
+     * @param proxyMethod the proxy method representing the page api.
      * @param settings the java settings.
      *
      * @return the paged metadata, or {@code null} if the {@code operation} is not a pageable operation.
      */
-    static PagingMetadata create(Operation operation, ProxyMethod proxyMethod, JavaSettings settings) {
+    static PagingMetadata create(Operation operation, ProxyMethod proxyMethod,
```

Comment:
given `PagingMetadata` consolidates everything about paging, I've updated it to also holds "maxPageSizeParameter" and "continuationTokenParameter"